src/Hydrodynamics/EnergyRadws.cpp

```diff
@@ -735,6 +665,77 @@ FLOAT EnergyRadws<ndim,ParticleType>::GetMuBar
 
 
 
+//=================================================================================================
+//  EnergyRadws::GetTemp()
+/// GetTemp returns the temperature of particle form it's density and specific internal energy.
+/// The value is taken from the EoS table, not via a direct calculation.
+//=================================================================================================
+template <int ndim, template <int> class ParticleType>
+FLOAT EnergyRadws<ndim,ParticleType>::GetTemp
+ (Particle<ndim> &part)
+{
+  FLOAT result = temp_min;
+
+  FLOAT logrho = log10(part.rho);
+  int idens = GetIDens(logrho);
+
+  if (part.u == 0.0) return result;
+
+  int temp_index = -1;
+  FLOAT du = 1E20;
+  for (int i = 0; i < ntemp; ++i) {
+    if (abs(part.u - eos_energy[idens][i]) < du) {
+      du = abs(part.u - eos_energy[idens][i]);
+      temp_index = i;
+    }
+  }
+
+  result = pow(10.0, eos_temp[temp_index]);
+
+  // Limit temperature to 5K (CMB)
+  if (result < temp_min) {
+    result = temp_min;
+    temp_index = GetITemp(temp_min);
+  }
+
+  part.mu_bar = eos_mu[idens][temp_index];
+  part.gamma = eos_gamma[idens][temp_index];
+
+  return result;
+}
+
+//=================================================================================================
+//  EnergyRadws::GetCol2()
+/// GetTemp returns the square of the mass-weighted column density average of a particle
+/// pseudocloud. The standard RadWS (Stamatellos et al. 2007) method uses the gravitational
+/// potential and the density of the particle. The Lombardi et al. (2015) method instead
+/// uses the hydrodynamical acceleration of a particle as well as its pressure.
+//=================================================================================================
+template <int ndim, template <int> class ParticleType>
+FLOAT EnergyRadws<ndim,ParticleType>::GetCol2
+ (Particle<ndim> &part)
+{
+  if (!lombardi) {
+    assert(part.gpot_hydro <= part.gpot);
+    return fcol2 * part.gpot_hydro * part.rho;
+  }
+  else {
+    FLOAT mag_da = 0.0, P = 0.0;
+
+    FLOAT tot_a = 0.0, tot_atree = 0.0;
+    for (int k = 0; k < ndim; ++k) {
+      tot_a += part.a[k] * part.a[k];
+      tot_atree += part.atree[k] * part.atree[k];
+    }
+    FLOAT mag_a = sqrt(tot_a);
+    FLOAT mag_atree = sqrt(tot_atree);
+    mag_da = mag_a - mag_atree;
```

Comment:
I suspect that this line is incorrect because |a_hydro|^2 = |a_tot - a_tree|^2, not = |a_tot| - |a_tree|^2,